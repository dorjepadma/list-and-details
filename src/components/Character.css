.Character {
  height: 60%;
  background-color: rgb(160, 160, 221);
  color: white;
  border: black;
  }

  img {
    display: block;
    margin: auto;
    width: 70%;
    height: 70%
  }
  
  h3 {
    text-align: center;
  }

  p {
    padding: 1rem 2rem;
    line-height: 1.5rem;
  }
