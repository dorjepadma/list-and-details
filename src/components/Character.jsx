import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

const Character = ({ image, name, description }) => (
  <section className={styles.Character}>
    <img src={image} alt={name} />
    <h3>{name}</h3>
    <p>{description}</p>
  </section>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string
};

export default Character;
