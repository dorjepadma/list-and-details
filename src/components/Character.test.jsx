import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character image="An Image" name="Air Bender" description="Drink Good Tea" />);
    expect(wrapper).toMatchSnapshot();
  });
});
