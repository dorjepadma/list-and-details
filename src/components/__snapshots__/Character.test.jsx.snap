// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<section
  className="Character"
>
  <img
    alt="Air Bender"
    src="An Image"
  />
  <h3>
    Air Bender
  </h3>
  <p>
    Drink Good Tea
  </p>
</section>
`;
